### ISSUE TITLE
Global marks not cleared correctly

### ISSUE THREAD

**Issue by sigzegv** [kshenoy/vim-signature#162]
When using 2 files on a splited vim window, if I set a global mark on the first file, and then set the same global mark on the second pane, mark on the first pane is not cleared in the gutter (while global mark is effectively removed from first file).

**Comment by kshenoy** [comment]
TLDR version:
@sigzegv Marks are updated only when you enter the buffer. I did it this way to keep the implementation simple.

RTFM version:
When you add a global mark to the second buffer, I'd have to figure out if it exists in another file, if that file is open in a buffer and clear the sign for it **before** I place the new mark or I lose information about where the mark was placed previously. Furthermore, if there are multiple global marks strewn about, every time a global mark is placed I'd have to do each of the above steps. 
In conclusion, all of this is possible and maybe not too difficult to implement. However, I chose to keep it simple. Updating only when the buffer is entered allows doing all the updates at once.

**Comment by sigzegv** [comment]
Yes I didn't clarified it, if you move back the gutter is cleaned correctly.
